Sources/DeepLinking/Route+Component.swift

```diff
@@ -2,70 +2,68 @@ import Foundation
 
 public extension Route {
 
+    /// A Route's path component.
     enum Component: Hashable {
-        case empty // for default handlers, e.g.: /home
+
+        /// A constant path component.
         case constant(String)
-        case variable(String?) // variables and wildcard (*)
 
+        /// A dynamic parameter component, represented as `:parameterName`
+        case parameter(String)
+
+        /// A dynamic parameter component with discarded value, represented as `*`.
+        case wildcard
+
+        /// A dynamic parameter component that matches one or multiple values, consuming the remaining path.
+        /// Represented as `**` with discarded value, or as `**parameterName` to capture the value.
+        case catchAll(String?)
+
+        /// Instantiates a new path component from a given string value.
+        ///
+        /// - Parameter component: The string value.
         public init(component: String) {
+
             precondition(component.contains("/") == false, "💥 Path components can't have any \"/\" characters!")
 
             switch component.first {
             case ":"?:
-                let index = component.index(component.startIndex, offsetBy: 1)
-                let parameterName = String(component[index...])
+                let parameterIndex = component.index(component.startIndex, offsetBy: 1)
+                let parameterName = String(component[parameterIndex...])
 
-                assert(parameterName.isEmpty == false, "🔥 Path component's parameter name is empty!")
+                precondition(parameterName.isEmpty == false, "🔥 Path component's parameter name is empty!")
```

Comment:
Should we really crash the program on -O builds? 🤔

Reply:
Hmm, that's a fair point... My idea was that these sort of invalid (route) components should be caught during development/testing and fail hard, however they might indeed slip into optimized builds and crash the app on scenarios where recovery is actually possible (e.g. by not allowing the route to be created on the router).

I think a less harsh approach would be to create a new error type to be thrown by the `Route.Component` initializer, which can then be mapped to an `invalidRoute` error on the router. What do you think?

Reply:
I have mixed feelings on the approach here, because this represent a programming error, and as you said, this should be caught during development. With this in mind, I would go for an assertion.

On the other hand, one might want to create routes dynamically on runtime, I suppose. If so, your idea for a new error type is better because allows one to handle the failure.

I'm fine with both approaches, but after thinking about the subject while writing this comment, I'm more inclined for the new error type you suggested.

Reply:
Yeah, my train of thought was essentially the same.

I've made an experiment with a new custom error, but it then "breaks" `ExpressibleByStringLiteral` on `Route.Component` 😭. However, since it's mostly useful on the unit tests, we can simply move it to that target and use a `try!` underneath 😇